src/function/scalar/list/list_contains.cpp

```diff
@@ -0,0 +1,226 @@
+	if (child_vector.GetType().id() == LogicalTypeId::SQLNULL) {
+		SetResultFalse(result);
+		return;
+	}
+	VectorData child_data;
+	child_vector.Orrify(list_size, child_data);
+
+	VectorData list_data;
+	list.Orrify(count, list_data);
+	auto list_entries = (list_entry_t *)list_data.data;
+
+	result.SetVectorType(VectorType::FLAT_VECTOR);
+	auto result_entries = FlatVector::GetData<bool>(result); // Create a vector of bool
+	auto &result_validity = FlatVector::Validity(result);
+
+	for (idx_t i = 0; i < count; i++) {
+		auto list_index = list_data.sel->get_index(i);
+
+		if (!list_data.validity.RowIsValid(list_index)) {
+			result_validity.SetInvalid(i);
+			continue;
+		}
+		result_entries[list_index] = false;
+
+		const auto &entry = list_entries[list_index];
+		auto source_idx = child_data.sel->get_index(entry.offset);
+		auto child_value = FlatVector::GetData<T>(child_vector);
+
+		for (idx_t child_idx = 0; child_idx < entry.length; child_idx++) {
+			auto value_idx = source_idx + child_idx;
+			if (!child_data.validity.RowIsValid(value_idx)) {
+				continue;
+			}
+			auto actual_value = child_value[value_idx];
+			if (actual_value == ((T *)value_data.data)[0]) {
+				result_entries[list_index] = true;
+				break; // Found value in list, no need to look further
+			}
+		}
+	}
+	return;
+}
+
+static void ListContainsFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	switch (args.data[1].GetType().InternalType()) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		TemplatedListContainsFunction<int8_t>(args, state, result);
+		break;
+	case PhysicalType::INT16:
+		TemplatedListContainsFunction<int16_t>(args, state, result);
+		break;
+	case PhysicalType::INT32:
+		TemplatedListContainsFunction<int32_t>(args, state, result);
+		break;
+	case PhysicalType::INT64:
+		TemplatedListContainsFunction<int64_t>(args, state, result);
+		break;
+	case PhysicalType::INT128:
+		TemplatedListContainsFunction<hugeint_t>(args, state, result);
+		break;
+	case PhysicalType::UINT8:
+		TemplatedListContainsFunction<uint8_t>(args, state, result);
+		break;
+	case PhysicalType::UINT16:
+		TemplatedListContainsFunction<uint16_t>(args, state, result);
+		break;
+	case PhysicalType::UINT32:
+		TemplatedListContainsFunction<uint32_t>(args, state, result);
+		break;
+	case PhysicalType::UINT64:
+		TemplatedListContainsFunction<uint64_t>(args, state, result);
+		break;
+	case PhysicalType::FLOAT:
+		TemplatedListContainsFunction<float>(args, state, result);
+		break;
+	case PhysicalType::DOUBLE:
+		TemplatedListContainsFunction<double>(args, state, result);
+		break;
+	case PhysicalType::VARCHAR:
+		TemplatedListContainsStringFunction<string_t>(args, state, result);
+		break;
+	case PhysicalType::LIST:
+		throw NotImplementedException("This function has not yet been implemented for nested types");
+	default:
+		throw InvalidTypeException(args.data[1].GetType().id(), "Invalid type for List Contains");
```

Comment:
We can check whether nested types have been supplied in `ListContainsBind` already, that way the errors happens before the query has started executing. This should also be a `NotImplementedException`, because I don't think any type should be invalid for this function